nitrocli/src/pinentry.rs

```diff
@@ -21,7 +21,45 @@ use error::Error;
 use std::process;
 
 
-const CACHE_ID: &str = "nitrocli:user";
+/// PIN type requested from pinentry.  The available PIN types correspond to the PIN types used by
+/// the Nitrokey devices:  user and admin.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum PinType {
+  /// The admin PIN.
+  Admin,
+  /// The user PIN.
+  User,
+}
+
+
+const CACHE_ID_PREFIX: &str = "nitrocli";
+const CACHE_ID_SUFFIX_ADMIN: &str = "admin";
+const CACHE_ID_SUFFIX_USER: &str = "user";
+
+
+fn get_cache_id(pin_type: PinType) -> String {
```

Comment:
I feel like those three functions should belong into an `impl` for `PinType`. Also, can you change the return type to `&'static str`? Lastly, I believe the Rust community mostly has reached consensus on not prefixing getters with `get_`. E.g.,

```rust
impl PinType {
  fn cache_id(self) -> &'static str {
    match self {
      PinType::Admin => "nitrocli:user",
      PinType::User => "nitrocli:admin",
    }
  }
  // ...
}
```

Lastly, the project uses two spaces of indentation. I'd hope we can keep things consistent. Perhaps I find the time to set up a CI for the project, I feel as if it would help now.

Reply:
Fair points.  I agree that the code style should be consistent.  Besides the indentation width, should we stick to the [Rust style guidelines][]?

[Rust style guidelines]: https://doc.rust-lang.org/1.0.0/style/README.html

Reply:
> Fair points. I agree that the code style should be consistent. Besides the indentation width, should we stick to the [Rust style guidelines](https://doc.rust-lang.org/1.0.0/style/README.html)?

Sure thing.